src/server/routes/organisations/organisationSurveyController.js

```diff
@@ -56,24 +57,39 @@ const getOrganisationSurveys = async (req, res) => {
 const createOrganisationSurvey = async (req, res) => {
   const { user } = req
   const { code } = req.params
-  const { name, startDate, endDate, studentNumbers: initialStudentNumbers, teacherIds: initialTeacherIds } = req.body
+  const {
+    name,
+    startDate,
+    endDate,
+    studentNumbers: initialStudentNumbers,
+    teacherIds: initialTeacherIds,
+    courseIds: initialCourseIds,
+  } = req.body
 
   const { organisation, hasAdminAccess } = await getAccessAndOrganisation(user, code, {
     admin: true,
   })
 
   if (!hasAdminAccess) throw new ApplicationError('Only organisation admins can create organisation surveys', 403)
 
+  const studentNumbersFromCourseIds = await getStudentNumbersFromCourseIds(initialCourseIds)
+
   // Remove duplicates from studentNumbers and teacherIds
-  const studentNumbers = [...new Set(initialStudentNumbers)]
+  const studentNumbers = [
+    ...new Set([...initialStudentNumbers, ...studentNumbersFromCourseIds.filter(n => n !== null)]),
```

Comment:
You can use `...studentNumbersFromCourseIds.filter(n => Boolean(n))` or even `...studentNumbersFromCourseIds.filter(n => n)` here instead